model-test/src/main/java/org/jboss/as/model/test/ModelTestModelControllerService.java

```diff
@@ -519,7 +524,8 @@ public void start(StartContext context) throws StartException {
 
     public void waitForSetup() throws Exception {
         latch.await();
-        if (error != null) {
+        // Don't throw in case boot failure is expected
+        if (error != null && !error.equals(unknownBootFailure)) {
```

Comment:
I'm not sure why I worded it like that. The issue was that `OperationContextImpl.validateCapabilities` doesn't specify a reason if the server fails due to missing capabilities, and that should be ignored when the server is booting in ADMIN_ONLY. It might make sense to just perform the admin check here, but I wasn't sure at the time if that would break anything.

I just avoided that conflict by not actually throwing the error, but still saving an "unknown boot failure" exception for `getBootError`.

Reply:
Did some testing, it seems a few test cases don't set the `ADMIN_ONLY` running mode, and fail because they don't attempt to catch the exception. Besides the current change, I could either add exception handling to those test cases, or set them to `ADMIN_ONLY`. Here are the tests:

```suggestion
        // Don't throw in case boot failure is expected
        if (error != null && getRunningMode() != RunningMode.ADMIN_ONLY) {
```

This change resulted in test failures from `RemotingLegacySubsystemTestCase.testSubsystemWithBadConnectorProperty()` and most of the cases in `ValidateOperationsTestCase`.

```suggestion
        // Don't throw in case boot failure is expected
        if (error != null) {
```

The original code caused test failures in `RemotingLegacySubsystemTestCase.testSubsystemWithBadConnectorProperty()` as well, along with:

- `RemotingSubsystemTestCase.testCliAddProperty()`
- `StandaloneDeploymentUnicityTestCase.testIncorrectDeployments()`
- `DomainServerGroupTestCase.testAddServerGroupBadProfile()`